src/Tickets/Admin/All_Tickets/List_Table.php

```diff
@@ -498,31 +567,111 @@ public function modify_sort_args( $args ): array {
 				break;
 		}
 
-		$order         = tribe_get_request_var( 'order', 'desc' );
-		$args['order'] = strtoupper( $order );
+		$args['order'] = tribe_get_request_var( 'order', self::$default_sort_order );
 
 		return $args;
 	}
 
 	/**
-	 * Get the query args for the list table.
+	 * Modify the filter arguments.
 	 *
 	 * @since TBD
 	 *
+	 * @param array $args The arguments used to query the tickets for the All Tickets Table.
+	 *
 	 * @return array
 	 */
-	public function get_query_args(): array {
+	public function modify_filter_args( $args ) {
+		$filter = tribe_get_request_var( self::$status_key, self::$default_status );
+
+		if ( 'all' === $filter ) {
+			return $args;
+		}
+
+		switch ( $filter ) {
+			case 'active':
+				$args['meta_query'][] = [
+					'key'     => '_ticket_start_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '<=',
+					'type'    => 'DATETIME',
+				];
+				$args['meta_query'][] = [
+					'key'     => '_ticket_end_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				];
+				break;
+			case 'past':
+				$args['meta_query'][] = [
+					'key'     => '_ticket_end_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '<=',
+					'type'    => 'DATETIME',
+				];
+				break;
+			case 'upcoming':
+				$args['meta_query'][] = [
+					'key'     => '_ticket_start_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				];
+				break;
+			case 'discounted':
+				$args['meta_query'][] = [
+					'key'     => '_sale_price',
+					'compare' => 'EXISTS',
+				];
+				$args['meta_query'][] = [
+					'key'     => '_sale_price_start_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '<',
+					'type'    => 'DATETIME',
+				];
+				$args['meta_query'][] = [
+					'key'     => '_sale_price_end_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				];
+				break;
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Get the query args for the list table.
+	 *
+	 * @since TBD
+	 *
+	 * @return array|bool
+	 */
+	public function get_query_args() {
 		$current_page = $this->get_pagenum();
 		$per_page     = $this->get_items_per_page( $this->per_page_option );
 
+		$provider_options = $this->get_provider_options();
+		$default_provider = empty( $provider_options ) ? '' : key( $provider_options );
+		$current_provider = tribe_get_request_var( self::$provider_key, $default_provider );
```

Comment:
could `tribe_get_request_var` return `''` instead of the ~$default_provider` if the var is set in the `$_REQUEST` as an empty string ? If yes, we should make sure we are not continuing with empty string as the `$current_provider`

Reply:
Good idea. We should probably check the value against the array of providers to make sure it's valid, too. Thanks!